metaflow/plugins/cards/card_decorator.py

```diff
@@ -0,0 +1,192 @@
+        else:
+            self.card_options = self.attributes["options"]
+
+        self._flow_datastore = flow_datastore
+        self._environment = environment
+
+    def task_pre_step(
+        self,
+        step_name,
+        task_datastore,
+        metadata,
+        run_id,
+        task_id,
+        flow,
+        graph,
+        retry_count,
+        max_user_code_retries,
+        ubf_context,
+        inputs,
+    ):
+        self._task_datastore = task_datastore
+        self._metadata = metadata
+
+    def task_finished(
+        self, step_name, flow, graph, is_task_ok, retry_count, max_user_code_retries
+    ):
+        if not is_task_ok:
+            return
+        runspec = "/".join([current.run_id, current.step_name, current.task_id])
+        self._run_cards_subprocess(runspec)
+
+    @staticmethod
+    def _options(mapping):
+        for k, v in mapping.items():
+            if v:
+                k = k.replace("_", "-")
+                v = v if isinstance(v, (list, tuple, set)) else [v]
+                for value in v:
+                    yield "--%s" % k
+                    if not isinstance(value, bool):
+                        yield to_unicode(value)
+
+    def _create_top_level_args(self):
+
+        top_level_options = {
+            "quiet": True,
+            "metadata": self._metadata.TYPE,
+            "coverage": "coverage" in sys.modules,
+            "environment": self._environment.TYPE,
+            "datastore": self._flow_datastore.TYPE,
+            "datastore-root": self._flow_datastore.datastore_root,
+            # We don't provide --with as all execution is taking place in
+            # the context of the main processs
+        }
+        # FlowDecorators can define their own top-level options. They are
+        # responsible for adding their own top-level options and values through
+        # the get_top_level_options() hook.
+        for deco in flow_decorators():
+            top_level_options.update(deco.get_top_level_options())
+        return list(self._options(top_level_options))
+
+    def _run_cards_subprocess(self, runspec):
+        executable = sys.executable
+        cmd = [
+            executable,
+            sys.argv[0],
+        ]
+        cmd += self._create_top_level_args() + [
+            "card",
+            "create",
+            runspec,
+            "--type",
+            self.attributes["type"],
+            # Add the options relating to card arguments.
+            # todo : add scope as a CLI arg for the create method.
+        ]
+        if self.card_options is not None and len(self.card_options) > 0:
+            cmd += ["--options", json.dumps(self.card_options)]
+        # set the id argument.
+
+        if self.attributes["timeout"] is not None:
+            cmd += ["--timeout", str(self.attributes["timeout"])]
+
+        self._run_command(cmd, os.environ)
+        # do nothing on failure as we create an error card
```

Comment:
if the subprocess has a hard crash, nothing is shown currently - not even an error card. You could handle soft crashes quietly with the error card, which can return 0, but if the subprocess fails hard, you could print an error on stderr with the error code